### PR TITLE
#24 API for go: integration test implemented

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #24 I've implemented IT case. Please, note that `com.artipie.goproxy.GoSliceITCase.Fake` is not supposed to stay in the repo forever, we will replace it with actual `Slice` implementation.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#25](https://github.com/artipie/go-adapter/issues/25) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/b341f217-fe77-488c-9ebc-2259665bbcf7); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/goproxy/GoSliceITCase.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+    /**
+     * Vertx instance.
+     */
+    private static final Vertx VERTX = Vertx.vertx();
+
+    /**
+     * Port to listen for go adapter.
+     */
+    private static final int DEF_PORT = 9080;
```
Comment (Vatavuk) [inline src/test/java/com/artipie/goproxy/GoSliceITCase.java:69]: @olenagerasimova this port might be occupied depending on a different machine where tests are executed. Let's add a method that will find available port and provide that port to the container and server.

File: src/test/java/com/artipie/goproxy/GoSliceITCase.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                    new Fake()
+                )
+            ),
+            GoSliceITCase.DEF_PORT
+        ).start();
+    }
+
+    @AfterAll
+    static void stopContainer() {
+        GoSliceITCase.VERTX.close();
```
Comment (Vatavuk) [inline src/test/java/com/artipie/goproxy/GoSliceITCase.java:111]: @olenagerasimova we should close `VertxSliceServer`

**Comment by olenagerasimova** [comment]
@Vatavuk thanks, updated, please, take a look

**Comment by olenagerasimova** [comment]
@g4s8 ping

File: pom.xml [hunk 3]
```diff
@@ -92,6 +87,16 @@ SOFTWARE.
...
     </dependency>
+    <dependency>
+      <groupId>com.artipie</groupId>
+      <artifactId>http</artifactId>
+      <version>0.7.4</version>
+    </dependency>
+    <dependency>
+      <groupId>com.artipie</groupId>
+      <artifactId>vertx-server</artifactId>
+      <version>0.2.2</version>
```
Comment (g4s8) [inline pom.xml:98]: @olenagerasimova I suppose it should be `test` dependency

**Comment by olenagerasimova** [comment]
@g4s8 thanks, corrected

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**